test_storage_s3_queue is flaky

Even after https://github.com/ClickHouse/ClickHouse/pull/77646 there are still

>E           helpers.client.QueryRuntimeException: Client failed! Return code: 210, stderr: Code: 210. DB::NetException: Connection refused (172.16.1.11:9000). (NETWORK_ERROR)

https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=78013&sha=5648b1310a55e47d3a61e03fc9510f6ff782803b&name_0=PR&name_1=Integration%20tests%20%28release%2C%202%2F4%29

@kssenii can you take a look when you have a minute?

https://s3.amazonaws.com/clickhouse-test-reports/PRs/77841/42886968f1af877d77394c74cf07e0bf9dcbda45//integration_tests_tsan_6_6/integration_run_test_storage_s3_queue_test_4_py_0.log

```
E           Exception: Sanitizer assert found for instance ==================
E           WARNING: ThreadSanitizer: data race (pid=2462)
E             Write of size 8 at 0x7250003046d8 by thread T638:
E               #0 std::__1::unique_ptr<ThreadFromGlobalPoolImpl<true, true>, std::__1::default_delete<ThreadFromGlobalPoolImpl<true, true>>>::reset[abi:ne190107](ThreadFromGlobalPoolImpl<true, true>*) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:290:20 (clickhouse+0x1b7cb690) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #1 std::__1::unique_ptr<ThreadFromGlobalPoolImpl<true, true>, std::__1::default_delete<ThreadFromGlobalPoolImpl<true, true>>>::operator=[abi:ne190107](std::__1::unique_ptr<ThreadFromGlobalPoolImpl<true, true>, std::__1::default_delete<ThreadFromGlobalPoolImpl<true, true>>>&&) build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:232:5 (clickhouse+0x1b7cb690)
E               #2 DB::ObjectStorageQueueMetadata::startup() build_docker/./src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp:173:32 (clickhouse+0x1b7cb690)
E               #3 DB::StorageObjectStorageQueue::startup() build_docker/./src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp:269:25 (clickhouse+0x1b89f8d6) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #4 DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0::operator()(DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./src/Databases/DatabaseOrdinary.cpp:467:24 (clickhouse+0x175e42be) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #5 decltype(std::declval<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&>()(std::declval<DB::AsyncLoader&>(), std::declval<std::__1::shared_ptr<DB::LoadJob> const&>())) std::__1::__invoke[abi:ne190107]<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&>(DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x175e42be)
E               #6 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&>(DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x175e42be)
E               #7 std::__1::__function::__default_alloc_func<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0, void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()[abi:ne190107](DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x175e42be)
E               #8 void std::__1::__function::__policy_invoker<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0, void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>>(std::__1::__function::__policy_storage const*, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x175e42be)
E               #9 std::__1::__function::__policy_func<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()[abi:ne190107](DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10e7fbe0) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #10 std::__1::function<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()(DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10e7fbe0)
E               #11 DB::LoadJob::execute(DB::AsyncLoader&, unsigned long, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./src/Common/AsyncLoader.cpp:172:5 (clickhouse+0x10e7fbe0)
E               #12 DB::AsyncLoader::worker(DB::AsyncLoader::Pool&) build_docker/./src/Common/AsyncLoader.cpp:947:18 (clickhouse+0x10e7fbe0)
E               #13 DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0::operator()() const build_docker/./src/Common/AsyncLoader.cpp:882:5 (clickhouse+0x10e80912) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #14 decltype(std::declval<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>(DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x10e80912)
E               #15 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>(DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x10e80912)
E               #16 std::__1::__function::__default_alloc_func<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0, void ()>::operator()[abi:ne190107]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x10e80912)
E               #17 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x10e80912)
E               #18 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10c1e8ae) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #19 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10c1e8ae)
E               #20 ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::worker() build_docker/./src/Common/ThreadPool.cpp:799:17 (clickhouse+0x10c1e8ae)
E               #21 decltype(*std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&>().*std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&, void>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x10c26de5) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #22 decltype(auto) std::__1::__apply_tuple_impl[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, 0ul>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<0ul>) build_docker/./contrib/llvm-project/libcxx/include/tuple:1354:5 (clickhouse+0x10c26de5)
E               #23 decltype(auto) std::__1::apply[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&) build_docker/./contrib/llvm-project/libcxx/include/tuple:1358:5 (clickhouse+0x10c26de5)
E               #24 ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()::operator()() build_docker/./src/Common/ThreadPool.h:312:13 (clickhouse+0x10c26de5)
E               #25 decltype(std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)()>()(std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>())) std::__1::__invoke[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x10c26d01) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #26 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x10c26d01)
E               #27 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>::operator()[abi:ne190107]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x10c26d01)
E               #28 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x10c26d01)
E               #29 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10c1af5f) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #30 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10c1af5f)
E               #31 ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::worker() build_docker/./src/Common/ThreadPool.cpp:809:17 (clickhouse+0x10c1af5f)
E               #32 decltype(*std::declval<ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>().*std::declval<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, void>(void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*&&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x10c23a9b) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #33 void std::__1::__thread_execute[abi:ne190107]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<2ul>) build_docker/./contrib/llvm-project/libcxx/include/__thread/thread.h:192:3 (clickhouse+0x10c23a9b)
E               #34 void* std::__1::__thread_proxy[abi:ne190107]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>>(void*) build_docker/./contrib/llvm-project/libcxx/include/__thread/thread.h:201:3 (clickhouse+0x10c23a9b)
E           
E             Previous read of size 8 at 0x7250003046d8 by thread T659:
E               #0 std::__1::unique_ptr<ThreadFromGlobalPoolImpl<true, true>, std::__1::default_delete<ThreadFromGlobalPoolImpl<true, true>>>::operator bool[abi:ne190107]() const build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:279:19 (clickhouse+0x1b7cb4b8) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #1 DB::ObjectStorageQueueMetadata::startup() build_docker/./src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp:172:10 (clickhouse+0x1b7cb4b8)
E               #2 DB::StorageObjectStorageQueue::startup() build_docker/./src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp:269:25 (clickhouse+0x1b89f8d6) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #3 DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0::operator()(DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./src/Databases/DatabaseOrdinary.cpp:467:24 (clickhouse+0x175e42be) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #4 decltype(std::declval<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&>()(std::declval<DB::AsyncLoader&>(), std::declval<std::__1::shared_ptr<DB::LoadJob> const&>())) std::__1::__invoke[abi:ne190107]<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&>(DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x175e42be)
E               #5 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&>(DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0&, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x175e42be)
E               #6 std::__1::__function::__default_alloc_func<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0, void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()[abi:ne190107](DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x175e42be)
E               #7 void std::__1::__function::__policy_invoker<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::DatabaseOrdinary::startupTableAsync(DB::AsyncLoader&, std::__1::unordered_set<std::__1::shared_ptr<DB::LoadJob>, std::__1::hash<std::__1::shared_ptr<DB::LoadJob>>, std::__1::equal_to<std::__1::shared_ptr<DB::LoadJob>>, std::__1::allocator<std::__1::shared_ptr<DB::LoadJob>>>, DB::QualifiedTableName const&, DB::LoadingStrictnessLevel)::$_0, void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>>(std::__1::__function::__policy_storage const*, DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x175e42be)
E               #8 std::__1::__function::__policy_func<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()[abi:ne190107](DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10e7fbe0) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #9 std::__1::function<void (DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&)>::operator()(DB::AsyncLoader&, std::__1::shared_ptr<DB::LoadJob> const&) const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10e7fbe0)
E               #10 DB::LoadJob::execute(DB::AsyncLoader&, unsigned long, std::__1::shared_ptr<DB::LoadJob> const&) build_docker/./src/Common/AsyncLoader.cpp:172:5 (clickhouse+0x10e7fbe0)
E               #11 DB::AsyncLoader::worker(DB::AsyncLoader::Pool&) build_docker/./src/Common/AsyncLoader.cpp:947:18 (clickhouse+0x10e7fbe0)
E               #12 DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0::operator()() const build_docker/./src/Common/AsyncLoader.cpp:882:5 (clickhouse+0x10e80912) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #13 decltype(std::declval<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>()()) std::__1::__invoke[abi:ne190107]<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>(DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x10e80912)
E               #14 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&>(DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x10e80912)
E               #15 std::__1::__function::__default_alloc_func<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0, void ()>::operator()[abi:ne190107]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x10e80912)
E               #16 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<DB::AsyncLoader::spawn(DB::AsyncLoader::Pool&, std::__1::unique_lock<std::__1::mutex>&)::$_0, void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x10e80912)
E               #17 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10c1e8ae) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #18 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10c1e8ae)
E               #19 ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::worker() build_docker/./src/Common/ThreadPool.cpp:799:17 (clickhouse+0x10c1e8ae)
E               #20 decltype(*std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&>().*std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&, void>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x10c26de5) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #21 decltype(auto) std::__1::__apply_tuple_impl[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, 0ul>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<0ul>) build_docker/./contrib/llvm-project/libcxx/include/tuple:1354:5 (clickhouse+0x10c26de5)
E               #22 decltype(auto) std::__1::apply[abi:ne190107]<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&)(), std::__1::tuple<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>&) build_docker/./contrib/llvm-project/libcxx/include/tuple:1358:5 (clickhouse+0x10c26de5)
E               #23 ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()::operator()() build_docker/./src/Common/ThreadPool.h:312:13 (clickhouse+0x10c26de5)
E               #24 decltype(std::declval<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)()>()(std::declval<ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>())) std::__1::__invoke[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:149:25 (clickhouse+0x10c26d01) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #25 void std::__1::__invoke_void_return_wrapper<void, true>::__call[abi:ne190107]<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&>(ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'()&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:224:5 (clickhouse+0x10c26d01)
E               #26 std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>::operator()[abi:ne190107]() build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:210:12 (clickhouse+0x10c26d01)
E               #27 void std::__1::__function::__policy_invoker<void ()>::__call_impl[abi:ne190107]<std::__1::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*>(void (ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::ThreadFromThreadPool*&&)::'lambda'(), void ()>>(std::__1::__function::__policy_storage const*) build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:610:12 (clickhouse+0x10c26d01)
E               #28 std::__1::__function::__policy_func<void ()>::operator()[abi:ne190107]() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:716:12 (clickhouse+0x10c1af5f) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #29 std::__1::function<void ()>::operator()() const build_docker/./contrib/llvm-project/libcxx/include/__functional/function.h:989:10 (clickhouse+0x10c1af5f)
E               #30 ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::worker() build_docker/./src/Common/ThreadPool.cpp:809:17 (clickhouse+0x10c1af5f)
E               #31 decltype(*std::declval<ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>().*std::declval<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)()>()()) std::__1::__invoke[abi:ne190107]<void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, void>(void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*&&)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*&&) build_docker/./contrib/llvm-project/libcxx/include/__type_traits/invoke.h:117:25 (clickhouse+0x10c23a9b) (BuildId: 4a160c0aae7301fdf3b5d5fc2cec3707e5318bfa)
E               #32 void std::__1::__thread_execute[abi:ne190107]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>&, std::__1::__tuple_indices<2ul>) build_docker/./contrib/llvm-project/libcxx/include/__thread/thread.h:192:3 (clickhouse+0x10c23a9b)
E               #33 void* std::__1::__thread_proxy[abi:ne190107]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void (ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool::*)(), ThreadPoolImpl<std::__1::thread>::ThreadFromThreadPool*>>(void*) build_docker/./contrib/llvm-project/libcxx/include/__thread/thread.h:201:3 (clickhouse+0x10c23a9b)

```

--- a/src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp
+++ b/src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp
@@ -156,6 +156,10 @@ ObjectStorageQueueMetadata::~ObjectStorageQueueMetadata()
 
 void ObjectStorageQueueMetadata::startup()
 {
+    if (startup_called)
+        return;
+    startup_called = true;
+
     if (!task
         && mode == ObjectStorageQueueMode::UNORDERED
         && (table_metadata.tracked_files_limit || table_metadata.tracked_files_ttl_sec))

--- a/src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.h
+++ b/src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.h
@@ -176,6 +176,7 @@ private:
     LoggerPtr log;
 
     std::atomic_bool shutdown_called = false;
+    std::atomic_bool startup_called = false;
     BackgroundSchedulePoolTaskHolder task;
 
     class LocalFileStatuses;

